### ISSUE TITLE
missing-parameter false positive for a method call on a pcall() result

### ISSUE THREAD

**Issue by ligurio** [EmmyLuaLs/emmylua-analyzer-rust#1248]
Imagine a file `pcall_unpack.lua` with the following Lua chunk:

```lua
local ok, err = pcall(function() error('boom') end)
if not ok then
    local x = err:unpack()
    print(x)
end
```

and the following config `.emmyrc.json` for the emmylua_check: `{ "runtime": { "version": "LuaJIT" } }`.

Run the command: `emmylua_check --config .emmyrc.json pcall_unpack.lua`
The command reports:
```
---  [2 warnings]
warning: The current Lua version Lua JIT is not accessible; expected >= Lua 5.3. [access-invisible]
  -->: :3:19

  2 │ if not ok then
  3 │     local x = err:unpack()
  4 │     print(x)

warning: expected 2 parameters but found 1. missing parameter: s [missing-parameter]
  -->: :3:26

  2 │ if not ok then
  3 │     local x = err:unpack()
  4 │     print(x)



Summary
  2 warnings

Check completed with warnings
Check finished
```

It is expected no diagnostics.

It seems the second return value of `pcall()` is typed as string, so `err:unpack()` is resolved to the Lua 5.3 `string.unpack(format, s, pos)` (missing `s`), even though `string.unpack()` is not accessible in LuaJIT (the access-invisible note confirms it).

Version: emmylua_check 0.24.0
